packages/features/users/components/UserTable/PlatformManagedUsersTable.tsx

```diff
@@ -0,0 +1,359 @@
+import { DeleteMemberModal } from "./DeleteMemberModal";
+import type { UserTableState, UserTableAction, PlatformManagedUserTableUser } from "./types";
+
+const initialState: UserTableState = {
+  changeMemberRole: {
+    showModal: false,
+  },
+  deleteMember: {
+    showModal: false,
+  },
+  impersonateMember: {
+    showModal: false,
+  },
+  inviteMember: {
+    showModal: false,
+  },
+  editSheet: {
+    showModal: false,
+  },
+};
+
+const initalColumnVisibility = {
+  select: true,
+  member: true,
+  role: true,
+  teams: true,
+  actions: true,
+};
+
+function reducer(state: UserTableState, action: UserTableAction): UserTableState {
+  switch (action.type) {
+    case "SET_CHANGE_MEMBER_ROLE_ID":
+      return { ...state, changeMemberRole: action.payload };
+    case "SET_DELETE_ID":
+      return { ...state, deleteMember: action.payload };
+    case "SET_IMPERSONATE_ID":
+      return { ...state, impersonateMember: action.payload };
+    case "INVITE_MEMBER":
+      return { ...state, inviteMember: action.payload };
+    case "EDIT_USER_SHEET":
+      return { ...state, editSheet: action.payload };
+    case "CLOSE_MODAL":
+      return {
+        ...state,
+        changeMemberRole: { showModal: false },
+        deleteMember: { showModal: false },
+        impersonateMember: { showModal: false },
+        inviteMember: { showModal: false },
+        editSheet: { showModal: false },
+      };
+    default:
+      return state;
+  }
+}
+
+type PlatformManagedUsersTableProps = {
+  oAuthClientId: string;
+};
+
+export function PlatformManagedUsersTable(props: PlatformManagedUsersTableProps) {
+  return (
+    <DataTableProvider>
+      <UserListTableContent {...props} />
+    </DataTableProvider>
+  );
+}
+
+function UserListTableContent({ oAuthClientId }: PlatformManagedUsersTableProps) {
+  const { t } = useLocale();
+
+  const tableContainerRef = useRef<HTMLDivElement>(null);
+  const [state, dispatch] = useReducer(reducer, initialState);
+  const [debouncedSearchTerm, setDebouncedSearchTerm] = useState("");
+  const [rowSelection, setRowSelection] = useState({});
+
+  const columnFilters = useColumnFilters();
+
+  const { data, isPending, hasNextPage, fetchNextPage, isFetching } =
+    trpc.viewer.organizations.listMembers.useInfiniteQuery(
+      {
+        limit: 30,
+        searchTerm: debouncedSearchTerm,
+        filters: columnFilters,
+        oAuthClientId,
+      },
+      {
+        getNextPageParam: (lastPage) => lastPage.nextCursor,
+        placeholderData: keepPreviousData,
+        enabled: !!oAuthClientId,
+      }
+    );
+
+  const totalDBRowCount = data?.pages?.[0]?.meta?.totalRowCount ?? 0;
+
+  //we must flatten the array of arrays from the useInfiniteQuery hook
+  const flatData = useMemo(
+    () => data?.pages?.flatMap((page) => page.rows) ?? [],
+    [data]
+  ) as PlatformManagedUserTableUser[];
+
+  const memorisedColumns = useMemo(() => {
```

Comment:
We can just name it `columns`

Reply:
Is it possible to maybe put the columns constant in a separate React component file to which we pass `isPending` as a prop? It's hard to read `UserListTableContent` component when the column code is so big and returns JSX and then the `UserListTableContent` component itself returns jsx.